src/VisualStudio/Core/Def/Library/ObjectBrowser/ObjectList.cs

```diff
@@ -747,8 +747,12 @@ protected override async Task GoToSourceAsync(uint index, VSOBJGOTOSRCTYPE srcTy
         {
             try
             {
-                using var token = _manager.AsynchronousOperationListener.BeginAsyncOperation(nameof(GoToSourceAsync));
-                using var context = _manager.OperationExecutor.BeginExecute(ServicesVSResources.IntelliSense, EditorFeaturesResources.Navigating, allowCancellation: true, showProgress: false);
+
+                var asynchronousOperationListener = _manager.ComponentModel.GetService<IAsynchronousOperationListenerProvider>().GetListener(FeatureAttribute.LibraryManager);
+                var operationExecutor = _manager.ComponentModel.GetService<IUIThreadOperationExecutor>();
+
+                using var token = asynchronousOperationListener.BeginAsyncOperation(nameof(GoToSourceAsync));
```

Comment:
📝 There is no asynchronous work started in this method which is not already captured by this method. It is unlikely that this token is needed here; if it's used at all it should be relocated to whichever caller started an operation that wasn't awaited.

Reply:
Yep, the caller is incorrect:
https://github.com/dotnet/roslyn/blob/69bdfb29ce4d027c5f5bf70f50be9bebcb5df1aa/src/VisualStudio/Core/Def/Library/AbstractObjectList.cs#L317-L318